### PR TITLE
✨(marsha) configure LRS variable in secret

### PR DESCRIPTION
**Pull request by lunika**
## Purpose

In order to use a LRS with marsha, we need to update the secret and add LRS related variables if they are defined.

## Proposal

- [x] Upgrade vault template with LRS variables commented.
- [x] Upgrade secret template using LRS variables if they are defined.

### CODE REVIEW

File: apps/marsha/templates/app/secret.yml.j2 [hunk 1]
```diff
@@ -17,3 +17,9 @@ data:
   DJANGO_JWT_SIGNING_KEY: "{{ MARSHA_VAULT.DJANGO_JWT_SIGNING_KEY | default('secret') | b64encode }}"
   DJANGO_SECRET_KEY: "{{ MARSHA_VAULT.DJANGO_SECRET_KEY | default('supersecret') | b64encode }}"
   DJANGO_SENTRY_DSN: "{{ MARSHA_VAULT.DJANGO_SENTRY_DSN | default('fakedsn') | b64encode }}"
+{% if MARSHA_VAULT.DJANGO_LRS_AUTH_TOKEN is defined%}
```
Comment (jmaupetit) [inline apps/marsha/templates/app/secret.yml.j2]:
```suggestion
{% if MARSHA_VAULT.DJANGO_LRS_AUTH_TOKEN is defined %}
```

File: apps/marsha/templates/app/secret.yml.j2 [hunk 2]
```diff
@@ -17,3 +17,9 @@ data:
...
+  DJANGO_LRS_AUTH_TOKEN: "{{ MARSHA_VAULT.DJANGO_LRS_AUTH_TOKEN }}"
+{% endif %}
+{% if MARSHA_VAULT.DJANGO_LRS_URL is defined%}
```
Comment (jmaupetit) [inline apps/marsha/templates/app/secret.yml.j2]:
```suggestion
{% if MARSHA_VAULT.DJANGO_LRS_URL is defined %}
```

File: apps/marsha/vars/vault/main.yml.j2 [hunk 3]
```diff
@@ -12,19 +12,25 @@ DJANGO_SECRET_KEY: {{ lookup('password', '/dev/null length=30') }}
...
 # DJANGO_CLOUDFRONT_ACCESS_KEY_ID: YourCloudfrontAccessKeyId
 # DJANGO_CLOUDFRONT_URL: https://yourCloudfrontUrl
-# DJANGO_CLOUDFRONT_PRIVATE_KEY: 
+# DJANGO_CLOUDFRONT_PRIVATE_KEY:
 
 # JWT Token
 DJANGO_JWT_SIGNING_KEY: {{ lookup('password', '/dev/null length=30') }}
+
+# Learning Locker variables
+# DJANGO_LRS_AUTH_TOKEN is the token provided by your LRS to authenticate in a request.
```
Comment (jmaupetit) [inline apps/marsha/vars/vault/main.yml.j2]:
```suggestion
# DJANGO_LRS_AUTH_TOKEN is the token provided by a LRS to authenticate requests
```

File: apps/marsha/vars/vault/main.yml.j2 [hunk 4]
```diff
@@ -12,19 +12,25 @@ DJANGO_SECRET_KEY: {{ lookup('password', '/dev/null length=30') }}
...
+# DJANGO_LRS_AUTH_TOKEN=Basic ThisIsABasicAuth
+# DJANGO_LRS_URL is the endpoint used to send xAPI statement.
```
Comment (jmaupetit) [inline apps/marsha/vars/vault/main.yml.j2]:
```suggestion
# DJANGO_LRS_URL is the endpoint used to send xAPI statements
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Added
+
+- Manage LRS variables in Marsha.
```
Comment (jmaupetit) [inline CHANGELOG.md]:
```suggestion
- Add support for LRS variables in Marsha.
```

**Review by jmaupetit — APPROVED** [review]

File: apps/marsha/vars/vault/main.yml.j2 [hunk 6]
```diff
@@ -12,19 +12,25 @@ DJANGO_SECRET_KEY: {{ lookup('password', '/dev/null length=30') }}
 # DJANGO_SENTRY_DSN: https://super:marsha@sentry.io/foo
 
 # FIXME uncomment this block and replace with your AWS credentials.
-# You can refer to the marsha documentation to know what are this variables: 
+# You can refer to the marsha documentation to know what are this variables:
```
Comment (jmaupetit) [inline apps/marsha/vars/vault/main.yml.j2]:
```suggestion
# You can refer to Marsha's documentation to know what are these variables:
```

File: apps/marsha/vars/vault/main.yml.j2 [hunk 7]
```diff
@@ -12,19 +12,25 @@ DJANGO_SECRET_KEY: {{ lookup('password', '/dev/null length=30') }}
...
 # https://github.com/openfun/marsha/blob/master/docs/env.md#amazon-web-services-related-settings
 # AWS
 # DJANGO_AWS_ACCESS_KEY_ID: yourAwsAccessKeyId
 # DJANGO_AWS_SECRET_ACCESS_KEY: yourAwsSecretAccessKey
 
 # FIXME uncomment this block and replace with your cloudfront credentials
-# You can refer to the marsha documentation to know what are this variables: 
+# You can refer to the marsha documentation to know what are this variables:
```
Comment (jmaupetit) [inline apps/marsha/vars/vault/main.yml.j2]:
```suggestion
# You can refer to Marsha's documentation to know what are these variables:
```

File: apps/marsha/vars/vault/main.yml.j2 (hunk 6 above)
Comment (lunika) [reply apps/marsha/vars/vault/main.yml.j2]: This is out of the scope of this PR :trollface:

Comment (jmaupetit) [reply apps/marsha/vars/vault/main.yml.j2]: Yes, I should have seen this earlier :stuck_out_tongue:

**Pull request merged**